src/TodoApp.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useRef, useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { useLocalStorage } from './hooks/LocalStorage';
+import { TodosContext } from './hooks/TodosContext';
+import { TodosFilter } from './components/TodosFilter';
+
+export const TodoApp: React.FC = () => {
+  const [storageTodos, setStorageTodos] = useLocalStorage();
+  const [todos, setTodos] = useState<Todo[]>(storageTodos);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activesCount = activeTodos.length;
+  const activesText = `${activesCount} items left`;
+  const allCompleted = completedTodos.length === todos.length;
+
+  const setNewTodos = (newTodos: Todo[]) => {
+    setTodos(newTodos);
+
+    return newTodos;
+  };
+
+  const handleNewSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const todoInput = inputRef.current;
+    const todoTitle = todoInput?.value.trim();
+
+    if (todoInput && todoTitle) {
+      const newTodo: Todo = {
+        id: +new Date(),
+        completed: false,
+        title: todoTitle,
+      };
+      const newTodos = [...todos, newTodo];
+
+      setTodos(newTodos);
+      setStorageTodos(newTodos);
+      todoInput.value = '';
+    }
+  };
```

Comment:
pls consider moving this logic to useEffect, so this won't be recalculated on every render

Reply:
Darek thinks, that extract logic to `useEffect` is [bad idea](https://mateacademyparttimepl.slack.com/archives/C04QUFZV2KF/p1697460943445979?thread_ts=1697457449.983769&cid=C04QUFZV2KF).